storage/migrations/08_evm_tokens_name_index.up.sql

```diff
@@ -0,0 +1,9 @@
+-- Add a trigram index for evm_tokens token names for search.
+
+BEGIN;
+
+CREATE EXTENSION pg_trgm;
```

Comment:
this breaks our wipe db script :skull:

Reply:
added some code to drop extensions too :skull: :skull:

Reply:
You could also just go with `CREATE EXTENSION IF NOT EXISTS pg_trgm`, and avoid the wiping. I'd prefer that, actually, for simplicity. It's extremely unlikely that we'll want to redefine the extension at some point :)

Reply:
Alternatively, I think it would also not be irresponsible to just omit the index. We have 2k EVM tokens in mainnet emerald. Even at 10k, scanning through their short names should be very fast.

Edit: I tried in prod mainnet now, with and without the index. The index _does_ get used. Make of the time difference what you will:
```sql
                                                                                                                                                                                                                                                                                 [0/21466]
oasisindexer=> explain analyze select count(*) from chain.evm_tokens where token_name ilike '%eth%';                                                                                                                                                                                      
                                                 QUERY PLAN                                                                                  
-------------------------------------------------------------------------------------------------------------                                
 Aggregate  (cost=68.17..68.18 rows=1 width=8) (actual time=1.491..1.493 rows=1 loops=1)                                                     
   ->  Seq Scan on evm_tokens  (cost=0.00..68.04 rows=53 width=0) (actual time=0.016..1.481 rows=58 loops=1)                                 
         Filter: (token_name ~~* '%eth%'::text)                                                                                              
         Rows Removed by Filter: 1865                        
 Planning Time: 0.325 ms                                              
 Execution Time: 1.544 ms         
(6 rows)                                                 
                                                                      
oasisindexer=> CREATE EXTENSION pg_trgm;                                                                                                     
CREATE EXTENSION                                                                                                                                                                                                                                                                          
oasisindexer=>                                                        
oasisindexer=> CREATE INDEX ix_evm_tokens_name ON chain.evm_tokens USING GIST (token_name gist_trgm_ops);                                    
CREATE INDEX                                                                                                                                                                                                                                                                              
oasisindexer=> CREATE INDEX ix_evm_tokens_symbol ON chain.evm_tokens USING GIST (symbol gist_trgm_ops);                                      
CREATE INDEX                             
oasisindexer=> explain analyze select count(*) from chain.evm_tokens where token_name ilike '%eth%';                                                                                                                                                                                      
                                                            QUERY PLAN                                                                       
-----------------------------------------------------------------------------------------------------------------------------------          
 Aggregate  (cost=51.69..51.70 rows=1 width=8) (actual time=0.455..0.456 rows=1 loops=1)                                                                                                                                                                                                  
   ->  Bitmap Heap Scan on evm_tokens  (cost=4.56..51.56 rows=53 width=0) (actual time=0.360..0.447 rows=58 loops=1)                                                                                                                                                                      
         Recheck Cond: (token_name ~~* '%eth%'::text)                                                                                                                                                                                                                                     
         Heap Blocks: exact=30                                 
         ->  Bitmap Index Scan on ix_evm_tokens_name  (cost=0.00..4.54 rows=53 width=0) (actual time=0.344..0.345 rows=58 loops=1)                                                                                                                                                        
               Index Cond: (token_name ~~* '%eth%'::text)
 Planning Time: 0.538 ms                            
 Execution Time: 0.651 ms   
```

Reply:
on 58 entries?

well I'll move this to a separate PR, let's hold off until it becomes slower

this query is going to run ~on keystroke, so

Reply:
oh but
```
{"args":[],"caller":"client.go:60","db":"indexer","err":"ERROR: cannot drop type gtrgm because extension pg_trgm requires it (SQLSTATE 2BP01)","level":"error","module":"postgres","msg":"Query","pid":188,"sql":"DROP TYPE public.gtrgm CASCADE;","time":"158.785µs","ts":"2023-06-27T18:11:29.493112027Z"}
```
with the current wipe code

Reply:
We can manually exclude `gtrgm` from the list of types to be wiped?

> on 58 entries?
58 that match `'%eth%'`, but ~2k entries total (`Rows Removed by Filter: 1865`)

Reply:
then there's a bunch of functions that it depends on